### PR TITLE
combine some pull requests

### PR DESCRIPTION
**Pull request by binsky08**
- integrates changes from
    - Change dots to ellipsis #81
    - fix autofill vault selection if the vault is active and selected as default #82
    - use uniform connect and response timeouts for all common api requests #83
    - Sort credentials by label automatically #37
- restrutcure settings
- fix endless loading dialog after deleting a credential
- migrate save and delete buttons to floating action buttons
- add confirm dialogue to delete a credential

### CODE REVIEW

File: app/src/main/java/es/wolfi/passman/API/Core.java [hunk 1]
```diff
@@ -48,7 +48,7 @@
     protected static String username;
     protected static String password;
     protected static String version_name;
-    protected static int version_number = 0;
+    protected static int versionNumber = 0;
```
Comment (animalillo) [inline app/src/main/java/es/wolfi/passman/API/Core.java:51]: This naming scheme is inconsistent with the rest of the class properties, i believe snake_case is more appropiate to visually distinguish them from methods.

File: app/src/main/java/es/wolfi/utils/CredentialLabelSort.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+    private final int method;
+
+    public CredentialLabelSort(int method) {
+        this.method = method;
+    }
+
+    @Override
+    public int compare(Credential left, Credential right) {
+        if (method == 1) {
```
Comment (animalillo) [inline app/src/main/java/es/wolfi/utils/CredentialLabelSort.java:21]: What does one and two mean? What's the default behaviour? I think this class could use a bit more documentation overall.

**Review by animalillo — COMMENTED** [review]
Some very silly and little changes that would be nice to have, also, is that submodule cryptopp used or is some leftover from some tests? i don't remember, but i don't think it's used, looks like native code that is not used anymore

File: app/src/main/java/es/wolfi/app/passman/CredentialViewAdapter.java [hunk 3]
```diff
@@ -73,6 +73,7 @@ public void onClick(View v) {
                 }
             }
         });
+        //if (holder.mItem.isHidden()) holder.mView.setVisibility(View.GONE);
```
[the remark this answers is not in the document]
Comment (binsky08) [reply app/src/main/java/es/wolfi/app/passman/CredentialViewAdapter.java:76]: It's not necessary, but it's a code comment from you.
thought that will have a reason.
should I remove it?

File: app/src/main/java/es/wolfi/passman/API/Core.java (hunk 1 above)
Comment (binsky08) [reply app/src/main/java/es/wolfi/passman/API/Core.java]: changed it to snake case with 1a24073c528a77cb43e1833ee8f09d18d26c6406

File: app/src/main/java/es/wolfi/utils/CredentialLabelSort.java (hunk 2 above)
Comment (binsky08) [reply app/src/main/java/es/wolfi/utils/CredentialLabelSort.java]: changed it to constants with a14e0e817e204c647a8bf6bd77a4ed4a4ee0cee6

**Review by animalillo — APPROVED** [review]

**Pull request merged**